orangecontrib/prototypes/widgets/owexplpred.py

```diff
@@ -0,0 +1,409 @@
+
+            diff = np.sum(f1 - f2)
+            expl[0, a] += diff
+
+            # update variance
+            steps[0, a] += self.batchSize
+            iterations_reached[0, a] += self.batchSize
+            d = diff - mu[0, a]
+            mu[0, a] += d/steps[0, a]
+            M2[0, a] += d*(diff - mu[0, a])
+            var[0, a] = M2[0, a] / (steps[0, a]-1)
+
+            if (callback()):
+                break
+
+            # exclude from sampling if necessary
+            neededIter = zSq * var[0, a] / (self.error**2)
+            if (neededIter <= steps[0, a]) and (steps[0, a] >= self.minIter) or (steps[0, a] > self.maxIter):
+                iterations_reached[0, a] = self.maxIter + 1
+
+
+        expl[0, :] = expl[0, :]/steps[0, :]
+
+        # creating return array
+        domain = Domain([self.atr_names], [
+                        ContinuousVariable('contributions'), ContinuousVariable('errors')])
+        table = Table.from_list(domain, np.asarray(
+            self.atr_names.values).reshape(-1, 1))
+        ordered = np.argsort(np.abs(expl[0]))[::-1]
+        print (table.Y[:,0].shape)
+        print (expl.T.shape)
+        table.Y[:,0] = expl.T[ordered][:,0]
+        table.Y[:,1] = np.sqrt( zSq * var[0,:] / steps[0,:])
+        table.X = table.X[ordered]
+        return classValue, table
+
+    def _get_predictions(self, inst, classValue):
+        if isinstance(self.data.domain.class_vars[0], ContinuousVariable):
+            # regression
+            return self.model(inst)
+        else:
+            # classification
+            predictions = (self.model(inst) == classValue) * 1
+            return predictions
+
+
+class OWExplainPred(OWWidget):
+
+    name = "Explain Predictions"
+    description = "Computes attribute contributions to the final prediction with an approximation algorithm for shapely value"
+    #icon = "iconImage.png"
+    priority = 200
+    gui_error = settings.Setting(5)
+    gui_p_val = settings.Setting(5)
+    resizing_enabled = False
+
+    class Inputs:
+        data = Input("Data", Table, default=True)
+        model = Input("Model", Model, multiple=False)
+        sample = Input("Sample", Table)
+
+    class Outputs:
+        explanations = Output("Explanations", Table)
+
+    class Warning(OWWidget.Warning):
+        empty_data = widget.Msg("Empty dataset.")
+        sample_too_big = widget.Msg("Too many samples to explain.")
+        selection_not_matching = widget.Msg(
+            "Domain of dataset and the sample to be explained does not match")
+
+    def __init__(self):
+        super().__init__()
+        self.data = None
+        self.model = None
+        self.toExplain = None
+        self.explanations = None
+
+        self._task = None
+        self._executor = ThreadExecutor()
+
+        self.dataview = QTableView(verticalScrollBarPolicy=Qt.ScrollBarAlwaysOn,
+                                   horizontalScrollBarPolicy=Qt.ScrollBarAlwaysOn,
```

Comment:
Not sure why the scrollbars should be always on...
The widget definitely needs to be bigger, and then I expect horizontal scrolling will not be needed.
Vertical scrolling just when the table is big seems fine to me?